src/core/symbology/qgsmarkersymbollayer.cpp

```diff
@@ -2458,18 +2459,24 @@ bool QgsSvgMarkerSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFa
 QRectF QgsSvgMarkerSymbolLayer::bounds( QPointF point, QgsSymbolRenderContext &context )
 {
   bool hasDataDefinedSize = false;
-  double scaledSize = calculateSize( context, hasDataDefinedSize );
-  scaledSize = context.renderContext().convertToPainterUnits( scaledSize, mSizeUnit, mSizeMapUnitScale );
+  double scaledWidth = calculateSize( context, hasDataDefinedSize );
```

Comment:
I don't think these changes correctly account for data defined symbol names...?

Reply:
Hmm, you are right. I guess there's no way around calling the svg cache to get height for data defined names/paths. 

This means to get proper anchoring, we'll have to do the same in renderPoint too. You're good with that?